src/openpgp.js

```diff
@@ -344,6 +344,65 @@ export function decryptSessionKey({ message, privateKey, password }) {
 }
 
 
+/////////////////////////////////////////////
+//                                         //
+//   Public key signing and verification   //
+//                                         //
+/////////////////////////////////////////////
+
+
+/**
+ * Signs a paublic key.
+ * @param  {Key} publicKey                 public key to be signed
+ * @param  {Key|Array<Key>} privateKeys    array of keys or single key with decrypted secret key data to sign public key
+ * @return {Promise<String|Key>}           Public key object in form:
+ *                                           { publicKey:Key, publicKeyArmored:String }
+ * @static
+ */
+export function signPublicKey({ publicKey, privateKeys }) {
+  checkKey(publicKey, 'publicKey');
+  privateKeys = toArray(privateKeys);
+
+  if (asyncProxy) { // use web worker if available
+    return asyncProxy.delegate('signPublicKey', { publicKey, privateKeys });
+  }
+
+  return execute(() => {
+
+    const signedPublicKey = publicKey.sign(privateKeys);
+
+    return {
+      publicKey: signedPublicKey,
+      publicKeyArmored: signedPublicKey.armor()
+    };
+
+  }, 'Error signing public key');
+}
+
+/**
+ * Verifies public key
+ * @param  {Key} publicKey               public key object with signatures
+ * @param  {Key|Array<Key>} publicKeys   array of publicKeys or single key, to verify signatures
+ * @return {Promise<Object>}             cleartext with status of verified signatures in the form of:
+ *                                         { signatures: [{ keyid:String, valid:Boolean|null }] }
+ * @static
+ */
+export function verifyPublicKey({ publicKey, publicKeys }) {
+  checkKey(publicKey, 'publicKey');
+  publicKeys = toArray(publicKeys);
+
+  if (asyncProxy) { // use web worker if available
+    return asyncProxy.delegate('verifyPublicKey', { publicKey, publicKeys });
+  }
+
+  return execute(() => ({
+
+    signatures: publicKey.verify(publicKeys)
+
+  }), 'Error verifying signed public key');
+}
+
+
```

Comment:
Like mentioned before, the changes to this file can be reverted. Signing and verifying public keys is a very specific use case that can be done by invoking a private api. We should try to keep the public api as clean as possible to make it easy for newcomers to use the library. Advanced use cases will require private api use anyway.